### PR TITLE
[ELF] Adjust --compress-sections to support compression level

### PR DESCRIPTION
**Pull request by MaskRay**
zstd excels at scaling from low-ratio-very-fast to
high-ratio-pretty-slow. Some users prioritize speed and prefer disk read
speed, while others focus on achieving the highest compression ratio
possible, similar to traditional high-ratio codecs like LZMA.

Add an optional `level` to `--compress-sections` (#84855) to cater to
these diverse needs. While we initially aimed for a one-size-fits-all
approach, this no longer seems to work.
(https://richg42.blogspot.com/2015/11/the-lossless-decompression-pareto.html)

When --compress-debug-sections is used together, make
--compress-sections take precedence since --compress-sections is usually
more specific.

Remove the level distinction between -O/-O1 and -O2 for
--compress-debug-sections=zlib for a more consistent user experience.

### CODE REVIEW

File: lld/docs/ReleaseNotes.rst [hunk 1]
```diff
@@ -29,6 +29,9 @@ ELF Improvements
 * ``--compress-sections <section-glib>=[none|zlib|zstd]`` is added to compress
   matched output sections without the ``SHF_ALLOC`` flag.
   (`#84855 <https://github.com/llvm/llvm-project/pull/84855>`_)
+* ``--compression-level`` is added to set the compression level
+  for ``--compress-debug-sections=[zlib|zstd]`` and ``--compress-sections``.
+  (`#90567 <https://github.com/llvm/llvm-project/pull/90567>`_)
```
Comment (smithp35) [inline lld/docs/ReleaseNotes.rst:34]: Worth mentioning that default compression level is now independent of linker optimization level (Z_BEST_SPEED)?

**Review by smithp35 — APPROVED** [review]
No objections from me. I agree that keeping the default independent of optimisation level is the right thing to do.

One minor suggestion for the release notes.

**Comment by MaskRay** [comment]
> No objections from me. I agree that keeping the default independent of optimisation level is the right thing to do.
> 
> One minor suggestion for the release notes.

Thanks. ISTM adjusting `--compress-sections` seems better (and matches `zlib:1`-style strings I found elsewhere).

It might be useful for `--compress-debug-sections` to get the optional level as well but its use case seems obsoleted by `--compress-sections`...

File: lld/docs/ReleaseNotes.rst (hunk 1 above)
Comment (MaskRay) [reply lld/docs/ReleaseNotes.rst:32]: thx for the suggestion. adopting

**Comment by Colibrow** [comment]
@MaskRay Hi, I'm wondering if there need to add option for lzma which aims for the best size.

**Pull request merged**